### PR TITLE
Add documentation for not primitive attributes

### PR DESCRIPTION
**Pull request by jiayihu**
Basically used the same code as posted in https://github.com/WebReflection/hyperHTML/issues/240#issuecomment-419079981, but without the stuff in the constructor. Honestly, I don't know the reason for using `self.html` and `return self`.

Is it something related to the [content of this slide](http://webreflection.github.io/talks/WCRemoteConf/#/14) of yours? However the other examples don't follow this pattern, so I left it out of this PR. The goal is to keep it simple for newcomers like myself, so if it's some browser edge-case shit, I'd ignore it.

P.S.
Fixed some grammar error on the way, thanks to Grammarly ❤️

### CODE REVIEW

File: src/static/hyperhtml/documentation/index.html [hunk 1]
```diff
@@ -630,10 +630,46 @@ <h3 id="essentials-6-2">The data Attribute</h3>
...
+              <p>
+                If you however find yourself restricted by <code>data</code>, big news for you: hyperHTML doesn't lock you in and it lets you free to implement it as follows for instance.
+              </p>
+              <pre><code class="javascript">
+              customElements.define(
+                'h-welcome',
+                class HyperWelcome extends HTMLElement {
+                  constructor(...args) {
+                    super(...args);
+                    this.html = hyperHTML.bind(self);
```
Comment (WebReflection) [inline src/static/hyperhtml/documentation/index.html]: when you invoke `super` in **any** ECMAScrfipt class, it could return a different instance that what you'd expect.

Accordingly, you want to use such instance and return it to the outer world.

```js
const self = super(...args);
self.html = hyperHTML.bind(self);
return self;
```

This is a universal pattern that makes polyfill also work. As example, document-register-element polyfill passes the element that is being upgraded, and the super call would take care of enriching such element with all prototype descriptor the custom class has.

That's the only way to "upgrade" an element.

If you want to simplify this because you have transpilers that will do the dance for you (i.e. Babel would do the same once transpiled) it's also OK, so for c clarity sake you either use `self` everywhere and return it, or you use `this` but you have to bind it, while you are binding `self` in there that is the `window` object, not the current instance.

TL:DR please drop `self` as bind or use the example code I've written.

File: src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+customElements.define(
+  'h-welcome',
+  class HyperWelcome extends HTMLElement {
+    constructor(...args) {
+      super(...args);
+      this.html = hyperHTML.bind(this);
+    }
```
Comment (WebReflection) [inline src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js]: I'd rather go:

```js
get html() { return this._html || (this._html = hyperHTML(this)); }
```

and drop the constructor dance. I also would drop the `connectedCallback` since it's not needed if you render when the component receive the user. It wouldn't know what to render otherwise.

File: src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    get user() { return this._user; }
+    set user(value) {
+      this._user = value;
+      this.render();
+    }
+
+    connectedCallback() { this.render(); }
+    render() {
+      return this.html`<h1>Hello, ${this._user.name}</h1>`;
```
Comment (WebReflection) [inline src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js]: Witch Custom Element you don't need to return anything. Just populate the element via `this.html` and drop the return.

File: src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    }
+  }
+);
+
+hyperHTML.bind(document.getElementById('root'))`
```
Comment (WebReflection) [inline src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js]: specially for one-off operation, you can use `hyperHTML` right away without `bind`

File: src/static/hyperhtml/documentation/index.html (hunk 1 above)
Comment (jiayihu) [reply src/static/hyperhtml/documentation/index.html]: Thanks for the explaination, I removed `self` in favor of `this` since it was a leftover of copy-paste.

File: src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js (hunk 3 above)
Comment (jiayihu) [reply src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js]: Yup I know this, but I'd prefer to keep the return to make it feel more "pure" and less side-effectish.

File: src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js (hunk 4 above)
Comment (jiayihu) [reply src/static/hyperhtml/examples/example/react/not-primitive-props/hyper.js]: Nice to know, but it's better to keep it consistent across examples to avoid confusing readers

**Comment by jiayihu** [comment]
Okay fixed the requested changes and replaced the constructor stuff with `get html() { return this._html || (this._html = hyperHTML.bind(this)); }`, although I would have preferred to keep `this.html = hyperHTML.bind(this)` for consistency.

Also it's `this._html = hyperHTML.bind(this)`, and not `this._html = hyperHTML(this)` for the same consistency reason. 

I believe it's better to have an intuitive documentation than performance-wise, but inconsistent, examples

File: src/static/hyperhtml/documentation/index.html [hunk 5]
```diff
@@ -643,8 +643,9 @@ <h3 id="essentials-6-2">The data Attribute</h3>
                 class HyperWelcome extends HTMLElement {
                   constructor(...args) {
                     super(...args);
```
Comment (WebReflection) [inline src/static/hyperhtml/documentation/index.html]: this is useless. Please drop the constructor as a whole

**Comment by jiayihu** [comment]
Done!
